examples/jsm/nodes/lighting/AnalyticLightNode.js

```diff
@@ -167,9 +167,25 @@ class AnalyticLightNode extends LightingNode {
 
 		light.shadow.updateMatrices( light );
 
+		const currentRenderTarget = renderer.getRenderTarget();
+		const currentRenderObjectFunction = renderer.getRenderObjectFunction();
+
+		renderer.setRenderObjectFunction( ( object, ...params ) => {
+
+			if ( object.castShadow === true ) {
```

Comment:
Maybe it would be simpler to do something like... `renderer.setRenderObjectFilter( object => object.castShadow === true )`? /ping @sunag

Reply:
I think `setRenderObjectFunction` is more handler-oriented than filter-oriented. If there were any changes I would suggest `setRenderObjectHandler`.

Reply:
Yeah, I certainly agree "handler" would be a better name -- but I just think maybe it would be better for a user-facing API to be simpler and don't provide full "handling" behavior, but only the "filtering" one.

Reply:
We have more applications this way, for example, such as adding `onBeforeShadow` and `onAfterShadow` soon.